lib/Sema/CSSimplify.cpp

```diff
@@ -7471,8 +7535,15 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifySubclassOfConstraint(
   // smaller constraints.
   if (auto *packType = type->getAs<PackType>()) {
     for (unsigned i = 0, e = packType->getNumElements(); i < e; ++i) {
-      addConstraint(ConstraintKind::SubclassOf, packType->getElementType(i),
-                    classType, locator.withPathElement(LocatorPathElt::PackElement(i)));
+      auto eltType = packType->getElementType(i);
+      if (auto *packExpansionType = eltType->getAs<PackExpansionType>()) {
```

Comment:
This feels like there should be a common place which knows how to split pack expansion types into per-element checks regardless of constraint kind, otherwise it there is a non-zero possibility to miss some places...